src/tapcell/src/tapcell.tcl

```diff
@@ -581,31 +581,27 @@ proc tapcell { args } {
                     set master [$db findMaster $cnrcap_nwin_master]
 
                     if { ![string match [$master getConstName] $cnrcap_nwin_master] } {
```

Comment:
This comparison (and all of the others like it) is not correct.
Here's an example of it failing:
% tapcell -endcap_cpp 2 -distance 120  -tapcell_master FILLCELL_X1x  -endcap_master FILLCELL_X1
tapcell -endcap_cpp 2 -distance 120  -tapcell_master FILLCELL_X1x  -endcap_master FILLCELL_X1
Running tapcell...
Step 1: Cut rows...
[INFO] Macro blocks found: 0
[INFO] #Original rows: 57
[INFO] #Cut rows: 0
Step 2: Insert endcaps...
invalid command name "NULL"

findMaster returns the string NULL if it fails. How do you expect [NULL getConstNme] to return a value?
It errors. Furthermore, you don't need to use "string match" on strings. You can just use != here.
So it should be:

if { $cnrcap_nwin_master != NULL } {

Reply:
Requested fixes added in the last commits